src/eradiate/scenes/surface/_dtm.py

```diff
@@ -0,0 +1,197 @@
+from __future__ import annotations
+
+import pathlib
+import typing as t
+
+import attr
+import numpy as np
+import pint
+import xarray as xr
+
+from ._core import Surface, surface_factory
+from ..bsdfs import BSDF
+from ..core import KernelDict, SceneElement
+from ..shapes import BufferMeshShape, PLYMeshShape, Shape, shape_factory
+from ...attrs import documented, get_doc, parse_docs
+from ...contexts import KernelDictContext
+from ...units import unit_context_config as ucc
+from ...units import unit_registry as ureg
+from ...util.misc import onedict_value
+
+
+@surface_factory.register(type_id="dtm")
+@parse_docs
+@attr.s
+class DTMSurface(Surface):
+    """
+    DTM surface [``dtm``]
+
+    A surface description for digital terrain models. Consists of a shape and an
+    associated BSDF.
+
+    This surface supports instantiation based on preprocessed geotiff files,
+    triangulated meshes in ply-format and analytical surfaces based on functions
+    which map x and y coordinates to an elevation value.
+
+    For details refer to the corresponding constructor methods.
+    """
+
+    id: t.Optional[str] = documented(
+        attr.ib(
+            default="terrain",
+            validator=attr.validators.optional(attr.validators.instance_of(str)),
+        ),
+        doc=get_doc(SceneElement, "id", "doc"),
+        type=get_doc(SceneElement, "id", "type"),
+        init_type=get_doc(SceneElement, "id", "init_type"),
+        default='"surface"',
+    )
+
+    shape: Shape = documented(
+        attr.ib(
+            converter=attr.converters.optional(shape_factory.convert),
+            validator=attr.validators.optional(
+                attr.validators.instance_of((BufferMeshShape, PLYMeshShape))
+            ),
+            kw_only=True,
+        ),
+        doc="Shape describing the surface.",
+        type=".BufferMeshShape or .PLYMeshShape",
+        init_type=".BufferMeshShape or .PLYMeshShape or dict",
+    )
+
+    # bsdf: BSDF = documented(
+    #     attr.ib(
+    #         factory=LambertianBSDF,
+    #         converter=bsdf_factory.convert,
+    #         validator=attr.validators.instance_of(BSDF),
+    #     ),
+    #     doc="The reflection model attached to the surface.",
+    #     type=".BSDF",
+    #     init_type=".BSDF or dict, optional",
+    #     default=":class:`LambertianBSDF() <.LambertianBSDF>`",
+    # )
+
+    @staticmethod
+    def _compute_vertex_index(x, y, len_y):
+        return x * len_y + y
+
+    @classmethod
+    # I don't know how to define the typehint for mesh_fname.
+    def from_ply(cls, mesh_fname, bsdf: BSDF):
```

Comment:
```suggestion
    def from_ply(cls, filename: PathLike, bsdf: BSDF):
```
where `PathLike` is imported from `eradiate.typing`.

But what is the point of this constructor, in the end?

Reply:
Hm I suppose you're right, when the user has a ply mesh already, they can call the class constructor and explicitly pass a `PLYMeshShape`